source/common/quic/envoy_quic_proof_verifier.h

```diff
@@ -12,7 +12,9 @@ namespace Quic {
 class EnvoyQuicProofVerifier : public EnvoyQuicProofVerifierBase {
 public:
   EnvoyQuicProofVerifier(Envoy::Ssl::ClientContextSharedPtr&& context)
-      : context_(std::move(context)) {}
+      : context_(std::move(context)) {
+    ASSERT(context_.get());
+  }
 
```

Comment:
Does this change belong here? Not opposed to it, just seems unrelated to the main change in this PR

Reply:
it was a follow-up to a change requested in another PR.
I can split it into its own thing, but didn't think it was worth the CI cost :-P

Reply:
Fine with me as long as we're aware that we're doing it!